### ISSUE TITLE
Enumeration over potentially null collection in ServiceBusDiagnosticsSource

### ISSUE THREAD

**Issue by maciejpitucha** [Azure/azure-service-bus-dotnet#460]
## Actual Behavior

Method `private void SetTags(Activity activity, IList<Message> messageList)` in class `ServiceBusDiagnosticSource`  (line 742) can potentially enumerate over null collection (`IList<Message> messageList`). Please see stack trace below:

It seems that no messages are lost nor they end up in DLQ, yet a lot of exceptions are thrown. 

````
System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Where[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Azure.ServiceBus.ServiceBusDiagnosticSource.SetTags(Activity activity, IList`1 messageList) in C:\Repos\azure-service-bus-dotnet\src\Microsoft.Azure.ServiceBus\ServiceBusDiagnosticsSource.cs:line 742
   at Microsoft.Azure.ServiceBus.ServiceBusDiagnosticSource.ReceiveStop(Activity activity, Int32 messageCount, Nullable`1 status, IList`1 messageList) in C:\Repos\azure-service-bus-dotnet\src\Microsoft.Azure.ServiceBus\ServiceBusDiagn
osticsSource.cs:line 227
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.<ReceiveAsync>d__65.MoveNext() in C:\Repos\azure-service-bus-dotnet\src\Microsoft.Azure.ServiceBus\Core\MessageReceiver.cs:line 393
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.ServiceBus.Core.MessageReceiver.<ReceiveAsync>d__63.MoveNext() in C:\Repos\azure-service-bus-dotnet\src\Microsoft.Azure.ServiceBus\Core\MessageReceiver.cs:line 323
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.Azure.ServiceBus.MessageReceivePump.<MessagePumpTaskAsync>d__11.MoveNext() in C:\Repos\azure-service-bus-dotnet\src\Microsoft.Azure.ServiceBus\MessageReceivePump.cs:line 64
````

## Versions  
- OS platform and version: Win 10
- .NET Version: .NET Core 2.1.200
- NuGet package version or commit ID: Microsoft.Azure.ServiceBus 3.0.0

**Comment by nemakam** [comment]
Duplicate of #459

**Comment by maciejpitucha** [comment]
fixed in #461
